### DISCUSSION TITLE
Some questions about cmd and cache of ee4

### DISCUSSION THREAD

**Discussion by nguyenphong2001** [EasyEngine/easyengine#1601] [General]
Hi guys. I'm new to ee4 so have a few questions about command.
1. ee3 supports fastCGI. And does ee4 support fastCGI? I see currently, ee4 -cache is using redis. Looks like redis cache is the only choice with ee4?
2. options --local-db and --with-local-redis mean?

Look forward to the help. Thanks

**Comment by dhsathiya** [comment]
1. EE4 only supports Redis Cache.
2. In EE4 without `--local-db` and `--with-local-redis` all the sites will be sharing same DB and Redis container. If you want a site with it's own Redis and DB containers you can use the flags mentioned.

**Comment by ssuess** [comment]
Related question (I think): I realize I didn't set the `--cache` flag when creating my sites, does this mean they have no cache other than what a plugin might provide? Is there an update command to enable cache after the site is created? (The docs make is seem that only proxy cache can be added/changed after create?)

**Comment by ssuess** [comment]
hm, after a bit more research this does not seem possible. But migrating sites to an ee site with cache enabled is significantly more complex (read: FAILS when I replace the wp-content folder and import my olddb.sql and then add letsencrypt) than when cache is not present (where the above process seems to work fine). Is there really no manual way to enable cache AFTER site creation and setup?

**Comment by dhsathiya** [comment]
Yes. There's no way to do it right now. I have worked on the POC and requested changes here: https://github.com/EasyEngine/easyengine/pull/1550.

I hope this will be there in next to next version of EE. :)

**Reply by ssuess** [reply]
Thanks for the update. In the meantime, is there a document somewhere that would describe how to do this manually? Or does it make more sense to just install a wp cache plugin in the meantime?

**Reply by ssuess** [reply]
or to turn on proxy caching?

**Reply by dhsathiya** [reply]
We have not documented it anywhere as of now. What I suggest is to create a site and migrate there for time being. Delete old site and then re-create it with `--cache` flag.
Or other option is to buy another VPS and then create on new VPS and migrate there.

**Reply by ssuess** [reply]
The problem I had before with this solution is that restoring the old db to the new cached site broke things. Are there special db entries that need to be preserved?

**Reply by dhsathiya** [reply]
Nope. Nothing at all. Instead of importing DB first, first, sync the code and then import the DB.
Also, check if the WP table prefix is the same or not.
To bypass the cache you can add a query parameter in the url example: `example.com\?a=b`

**Reply by ssuess** [reply]
So I tried to migrate one of my sites by doing the following:
1. I have existing, running domain at `mydomain.com`
2. I backup the db from `mydomain.com`
3. I create new subdomain in dns called `edit.mydomain.com`
4. I run the following ee command: `ee site create edit.mydomain.com --type=wp --cache --admin-user=myadminuser --admin-pass='myadminpass' --admin-email=myemail@mydomain.com --ssl=le --le-mail=mysslemail@mydomain.com`
5. I copy files using: `cp -R /opt/easyengine/sites/mydomain.com/app/htdocs/wp-content/ /opt/easyengine/sites/edit.mydomain.com/app/htdocs/` 
6. I connect to the db and restore the backed up db `mydomain_com` to the new db `edit_mydomain_com`
7. I change the wp_options table entries for home and siteurl to `edit.mydomain.com`
8. I login to `edit.mydomain.com` and do the following:
-- using plugin better search replace I replace any instance in the db of `/opt/easyengine/sites/mydomain.com` with `/opt/easyengine/sites/edit.mydomain.com`
-- none of  my plugins are activated, so I have to manually re-enable them all
-- however the wp redis plugin which I installed but did not activate before backing up old site IS already activated

After all of the above, the site does NOT cache or work properly:
-- changes made in admin (to themes, to general settings, etc) are not saved reliably (most of the time not)
-- purging page or full cache does NOT resolve items not saving
-- both admin parts AND front end are being cached??

My plan was to have a working site `edit.mydomain.com` and then once verifying that everything was working properly, delete my old domain `mydomain.com` and add an alias for it to the new cached site. But in the end, I just deleted `edit.mydomain.com`, as it was completely unusable. Can you please tell me what I am doing wrong and how to migrate my site differently so that I can keep all of the settings but still make use of redis caching? Thanks!

**Reply by dhsathiya** [reply]
We are not caching Admin part at all for page cache. You might be seeing the issue of settings not getting saved due to object cache.
You can try with `wp cache flush` command once you are done with search-replace.

Let me explain the steps in a detailed manner here.

1. You create another site - temp.example.com
2. You create db export with `wp db dump` command from source site - example.com
3. You rsync the wp-content of example.com and temp.example.com properly.
```shell
rsync -avhP /path/to/source/wp-content/ /path/to/destination/wp-content/
```
4. Verify the things are properly synced.
5. copy db to temp.example.com
6. run `wp db import` command
7. make changes in wp-config.php if any
8. run `wp search-replace "https://example.com" "https://temp.example.com" --all-tables`
9. run `wp cache flush`
10. Enable object cache and nginx-helper plugins if disabled

These are general steps I do. Nothing else and they work all the time.

**Reply by ssuess** [reply]
Thanks, this process worked!
